lib/rubocop/cop/rspec/redundant_predicate_matcher.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for redundant predicate matcher.
+      #
+      # @example
+      #   # bad
+      #   expect(foo).to be_exist(bar)
+      #   expect(foo).not_to be_include(bar)
+      #
+      #   # good
+      #   expect(foo).to exist(bar)
+      #   expect(foo).not_to include(bar)
+      #
+      class RedundantPredicateMatcher < Base
+        extend AutoCorrector
+
+        MSG = 'Use `%<good>s` instead of `%<bad>s`.'
+        RESTRICT_ON_SEND =
+          %i[be_all be_cover be_end_with be_eql be_equal
+             be_exist be_exists be_include be_match
+             be_respond_to be_start_with].freeze
+
+        def on_send(node)
+          return if node.parent.block_type? || !replacable_arguments?(node)
+
+          method_name = node.method_name.to_s
+          replaced = replaced_method_name(method_name)
+          add_offense(node, message: message(method_name,
+                                             replaced)) do |corrector|
+            corrector.replace(node.loc.selector, replaced)
+          end
+        end
+
+        private
+
+        def message(bad_method, good_method)
+          format(MSG, bad: bad_method, good: good_method)
+        end
+
+        def replacable_arguments?(node)
+          if node.method?(:be_all)
+            node.first_argument.send_type?
```

Comment:
Wondering if all send nodes are fine?
```ruby
expect(foo).to be_all be_odd
```
is. But
```ruby
let(:expected_foo_element_value) { 0 }
it do
  expect(foo).to be_all(expected_foo_element_value)
```
? It's also a send node, but can't be used as
```ruby
  expect(foo).to all expected_foo_element_value
```

It not only has to be a send, it also has to be a matcher. But we can't know it for sure in our static code checks if it is.

Reply:
Ah, I see. That's for sure. Thanks for catching this. I wonder if I should exclude `all` from the scope, since it's hard to check if it's a matcher and I think it's likely to generate a lot of false positives. What do you think?

Reply:
Yes, that’s something I’ve been thinking, too. Maybe ‘be_all’ can be ignored for autocorrection when there’s a send node? We already ignore the block form. I don’t think ‘be_all(true)’ eg is problematic.

Reply:
We make it so that it is ignored in the autocorrection. 
Maybe we might as well make it not an offense in the first place.